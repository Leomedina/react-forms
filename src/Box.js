import React from 'react';

const Box = ({ id, color, width, height }) => {
  const styling = {
    backgroundColor: color,
    width: `${width}px`,
    height: `${height}px`
  };

  return (
    <div style={styling}></div>
  )
};
export default Box;