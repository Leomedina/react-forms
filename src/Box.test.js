import React from "react";
import { render } from "@testing-library/react";
import Box from './Box';

describe("Smoke and snapshot tests", function () {
  it("should render without crashing", function () {
    const removeBox = () => null;
    const { id, color, width, height } = {
      id: 1, width: 30, height: 40, color: "green"
    }
    render(<Box id={id} color={color} width={width} height={height} removeBox={removeBox} />);
  });

  it("should match snapshot", function () {
    const removeBox = () => null;
    const { id, color, width, height } = {
      id: 1, width: 30, height: 40, color: "green"
    }
    const { asFragment } = render(<Box id={id} color={color} width={width} height={height} removeBox={removeBox} />);
    expect(asFragment()).toMatchSnapshot();
  });
});