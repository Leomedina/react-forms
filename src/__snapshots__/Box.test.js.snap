// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    style="background-color: green; width: 30px; height: 40px;"
  />
  <button>
    X
  </button>
</DocumentFragment>
`;
